pkg/irsa/eks/eks.go

```diff
@@ -0,0 +1,85 @@
+package eks
+
+import (
+	"context"
+	"time"
+
+	"github.com/giantswarm/backoff"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	capi "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/irsa-operator/pkg/aws/scope"
+	"github.com/giantswarm/irsa-operator/pkg/aws/services/eks"
+	"github.com/giantswarm/irsa-operator/pkg/aws/services/iam"
+	"github.com/giantswarm/irsa-operator/pkg/key"
+	ctrlmetrics "github.com/giantswarm/irsa-operator/pkg/metrics"
+)
+
+type Service struct {
+	Client client.Client
+	Scope  *scope.ClusterScope
+
+	IAM *iam.Service
+	EKS *eks.Service
+}
+
+func New(scope *scope.ClusterScope, client client.Client) *Service {
+	return &Service{
+		Scope:  scope,
+		Client: client,
+
+		EKS: eks.NewService(scope),
+		IAM: iam.NewService(scope),
+	}
+}
+func (s *Service) Reconcile(ctx context.Context) error {
+	s.Scope.Logger().Info("Reconciling AWSManagedCluster CR for IRSA")
+	oidcURL, err := s.EKS.GetEKSOpenIDConnectProviderURL(s.Scope.ClusterName())
+	if err != nil {
+		s.Scope.Logger().Error(err, "failed to fetch EKS OIDC issuer URL")
+		return err
+	}
+
+	// Fetch custom tags from Cluster CR
+	cluster := &capi.Cluster{}
+	err = s.Client.Get(ctx, types.NamespacedName{Namespace: s.Scope.ClusterNamespace(), Name: s.Scope.ClusterName()}, cluster)
```

Comment:
I think it would be better to add the custom tags to the cluster scope instead of fetching it from here. We use the cluster scope to group together all the values read from the CRs, building the cluster scope first, then just applying some business logic using the values inside the scope.